doc/adr/0004-nats-headers.md

```diff
@@ -92,3 +129,82 @@ Implemented and merged to master.
 ## Consequences
 
 Use of headers is possible.
+
+## Compatibility Across NATS Clients
+
+The following is a list of features to insure compatibility across NATS clients
+that support headers. Because the feature in Go client and nats-server leverage
+the Go implementation as described above, the API used will determine how header
+names are serialized.
```

Comment:
Again, the GO client is now simply case-sensitive. The set/get/etc.. work with whatever user's input they get. Also, keys are not touched when header is decoded from the wire, or changed when encoded when sending the message out.